drivers/hmis_csv_twenty_twenty/app/controllers/hmis_csv_twenty_twenty/importer_errors_controller.rb

```diff
@@ -9,12 +9,11 @@ class HmisCsvTwentyTwenty::ImporterErrorsController < ApplicationController
 
   def show
     importer_log = HmisCsvTwentyTwenty::Importer::ImporterLog.find(params[:id].to_i)
-    @base_name = File.basename(
-      importer_log.summary.keys.detect { |v| v == params[:file] },
-      '.csv',
-    )
     @import = GrdaWarehouse::ImportLog.find_by(importer_log_id: importer_log.id)
-    pattern = '%::' + @base_name.downcase
+
+    @filename = HmisCsvTwentyTwenty::Importer::Importer.importable_files_map.keys.detect { |v| v == params[:file] }
+    pattern = '%::' + HmisCsvTwentyTwenty::Importer::Importer.importable_files_map[@filename].downcase
+
```

Comment:
Can't we just lose the pattern now and just get the appropriate model name? and the do `HmisCsvTwentyTwenty::Importer::ImportError.arel_table[:source_type] => HmisCsvTwentyTwenty::Importer::Importer.importable_files_map[@filename].downcase`?

Reply:
I believe that it is possible for the module name to sometimes be from the look-asides, it would depend when the error was generated.